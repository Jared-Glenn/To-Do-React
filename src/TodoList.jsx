import React, { useState } from "react";
import Todo from "./Todo.jsx";
import NewTodoForm from "./NewTodoForm.jsx";
import { v4 as uuid } from 'uuid';

const TodoList = () => {
    const [ currentTodoList, setCurrentTodoList ] = useState([]);

    const addTodo = ({ item }) => {
        const newTodo = { id: uuid(), item: item };
        setCurrentTodoList(currentTodoList => [...currentTodoList, newTodo]);
    };
    
    const deleteTodo = (id) => {
            const removedTodoIndex = currentTodoList.findIndex(x => x.id === id);
            currentTodoList.splice(removedTodoIndex, 1);
            setCurrentTodoList(currentTodoList => [...currentTodoList]);
        }

    return (
        <div>
            <h2>What To Do Today:</h2>
            <NewTodoForm
                addTodo={addTodo}/>
                <ul>
                {currentTodoList.map((todo) => (
                <Todo
                    id={todo.id}
                    item={todo.item}
                    deleteTodo={() => deleteTodo(todo.id)}
                    />
                ))}
                </ul>
        </div>
    )
};

export default TodoList;