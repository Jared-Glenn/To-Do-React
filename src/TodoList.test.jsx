import { render, fireEvent, toBeInTheDocument } from "@testing-library/react";
import '@testing-library/jest-dom';
import TodoList from "./TodoList";

// Smoke Test
it('renders without crashing', () => {
    render(<TodoList />);
});

// Snapshot Test
it('matches the snapshot', () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", () => {
    const { getByText, getByLabelText, queryByTestId } = render(<TodoList />);
    const input = getByLabelText("Item:");
    const button = getByText("Add To Do Item");

    let todo = queryByTestId("todo");
    expect(todo).not.toBeInTheDocument();

    fireEvent.change(input, { target: { value: 'stuff'}});
    fireEvent.click(button);

    todo = queryByTestId("todo");
    expect(todo).toBeInTheDocument();
});