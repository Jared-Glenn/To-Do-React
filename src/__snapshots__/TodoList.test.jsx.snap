// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      What To Do Today:
    </h2>
    <form>
      <label
        for="item"
      >
        Item: 
      </label>
      <input
        id="item"
        name="item"
        placeholder="Item Name"
        type="text"
        value=""
      />
      <button>
        Add To Do Item
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
